src/pages/Product/Product.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import './Product.scss';
+import Modal from './Modal';
+import ProductDetail from './ProductDetail';
+
+const Product = () => {
+  const [product, setProduct] = useState([]);
+  const [count, setCount] = useState(1);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const navigate = useNavigate();
+  const params = useParams();
+  const productId = params.id;
+
+  const eventHandler = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  const { name, thumbnail_image_url, price, detail, stock } = product;
+
+  useEffect(() => {
+    fetch(`/data/productData.json`)
+      .then(response => response.json())
+      .then(result => setProduct(result.data[0]));
+  }, [productId]);
+
+  // const token = localStorage.getItem('token') || ''; login창 merge되면 쓸 코드
+
+  const toBuy = e => {
+    e.preventDefault();
+    const token =
+      'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjYsImV4cCI6MTY2MTQ0MzI2NSwiaWF0IjoxNjYxMDgzMjY1fQ.KmF-Jp46fdHKwxS01SJ8PtF5yD1SkQP8rwQFA6tU9rQ';
+    const x = true;
+
+    if (x === true) {
+      fetch(`http://10.58.0.117:3000/user/cart/product/${productId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        body: JSON.stringify({
+          quantity: count,
+        }),
+      }).then(res => {
+        if (res.ok) {
+          return navigate('/cart');
+        } else {
+          alert('로그인이 필요한 기능입니다');
+          navigate('/login');
+        }
+      });
+    }
+  };
+
+  const goToCart = e => {
+    e.preventDefault();
+    const token =
+      'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjM3LCJleHAiOjE2NjEyOTE1MDQsImlhdCI6MTY2MTI1NTUwNH0.hMs9yhxOnyZBohpzIVqcG1T6Aej4ZU9MGCEieTUcYiQ';
+
+    const x = true;
+    if (x === true) {
+      fetch(`http://10.58.0.117:3000/user/cart/product/${productId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        body: JSON.stringify({
+          quantity: count,
+        }),
+      }).then(res => {
+        if (res.ok) {
+          return setIsModalOpen(true);
+        } else if (res.status === 401) {
+          alert('재고');
+        } else {
+          alert('로그인이 필요한 기능입니다');
+          navigate('/login');
+        }
+      });
+    }
+  };
+
+  const countPlus = e => {
+    setCount(preCount =>
+      preCount < stock ? preCount + 1 : alert('재고 수량을 넘었습니다')
+    );
+  };
+
+  const countMinus = e => {
+    setCount(preCount => (preCount <= 0 ? 0 : (preCount = 1)));
```

Comment:
지금 로직이면, 감소버튼을 누르면 precCount-1 이 되는 게 아니라, 0 아니면 1이 될 것같아요.
count가 1이면 감소버튼을 누를 수 없게 함수를 리턴시키는 로직으로 구성해 보는건 어떨까요? (early return 패턴입니다.)
```suggestion
  const countMinus = () => {
    if (count < 2) {
      return;
    }
    setCount(preCount => preCount - 1);
  };
```

Reply:
감사합니다 수정완료했습니다!